### ISSUE TITLE
broken COPY in ONBUILD context

### ISSUE THREAD

**Issue by djalal** [hadolint/hadolint#573]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

hadolint should see "ONBUILD WORKDIR ..." has been set, and allow for a relative <dst> path in "ONBUILD COPY <src> <dst>"

### Actual behavior

check fails with 

```
-:7 DL3045 warning: `COPY` to a relative destination without `WORKDIR` set.
```

### Steps to reproduce the behavior

regression seems to be introduced first by version v1.23.0-4-g4db396c, version 1.23.0-3 works fine.

> docker run --rm -i hadolint/hadolint:v1.23.0-4-g4db396c

> cat Dockerfile
```
FROM python:3.9

ONBUILD WORKDIR /usr/src/app
ONBUILD COPY requirements.txt .
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.23.0-4-g4db396c
```

**Dockerfile** (if relevant)

```Dockerfile
FROM python:3.9

ONBUILD WORKDIR /usr/src/app
ONBUILD COPY requirements.txt .
```

**Additional environment details (OS, stack version, etc.)**

**Comment by m-ildefons** [comment]
Thanks for the bug report, @djalal 
I've created a fix and it should be available in `master` in a few days.
